frontend/src/components/recipe.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import './recipe.css';
+
+export default function Recipe(props) {
+
+    // ingredients list
+    const [ingredients, setIngredients] = useState(props.recipe.ingredients);
+    const [newIngredient, setNewIngredient] = useState('');
+    const [ingredientPos, setIngredientPos] = useState(ingredients.length);
+
+    // instructions list
+    const [instructions, setInstructions] = useState(props.recipe.instructions);
+    const [newInstruction, setNewInstruction] = useState('');
+    const [instructionPos, setInstructionPos] = useState(instructions.length);
+
+    const addIngredient = () => {
+        if (newIngredient.trim().length !== 0) {
+            const newIngredients = [...ingredients];
+            newIngredients.splice(ingredientPos > ingredients.length ? ingredients.length : ingredientPos, 0, newIngredient.trim());
+            setIngredients(newIngredients);
+            setNewIngredient('');
+            setIngredientPos(ingredients.length + 1);
+            console.log(newIngredient);
+        }
+    };
+
+    const addInstruction = () => {
+        if (newInstruction.trim().length !== 0) {
+            const newInstructions = [...instructions];
+            newInstructions.splice(instructionPos > instructions.length ? instructions.length : instructionPos, 0, newInstruction.trim());
+            setInstructions(newInstructions);
+            setNewInstruction('');
+            setInstructionPos(instructions.length + 1);
+            console.log(newInstruction);
+        }
+    };
+
+    const deleteIngredient = (index) => {
+        if (window.confirm("Are you sure you want to remove this?")) {
+            const newIngredients = [...ingredients];
+            newIngredients.splice(index, 1);
+            setIngredients(newIngredients);
+            setIngredientPos(ingredients.length - 1);
+        }
+    };
+
+    const deleteInstruction = (index) => {
+        if (window.confirm("Are you sure you want to remove this?")) {
+            const newInstructions = [...instructions];
+            newInstructions.splice(index, 1);
+            setInstructions(newInstructions);
+            setInstructionPos(instructions.length - 1);
+        }
+    };
+
+    return (
+        <main>
+            <div className="FlexContainer">
+                <div><img className="FlexImage" src={props.recipe.image} alt={props.recipe.description} /></div>
```

Comment:
Very small suggestion, but in the case the image doesn't work, having the alt be the same as the description, and placed right next to the actual description as well might look weird.